packages/compass-schema-validation/src/components/validation-states/validation-states.jsx

```diff
@@ -131,30 +132,37 @@ class ValidationStates extends Component {
    * @returns {React.Component} The component.
    */
   renderZeroState() {
-    if (this.props.isZeroState) {
-      return (
-        <div className={classnames(styles['zero-state-container'])}>
-          <ZeroGraphic />
-          <ZeroState header={HEADER} subtext={SUBTEXT}>
-            <div className={classnames(styles['zero-state-action'])}>
-              <div>
-                <TextButton
-                  className={`btn btn-primary btn-lg ${
-                    !this.isEditable() ? 'disabled' : ''
-                  }`}
-                  text="Add Rule"
-                  clickHandler={this.props.changeZeroState.bind(this, false)} />
-              </div>
-              <a
-                className={classnames(styles['zero-state-link'])}
-                onClick={this.props.openLink.bind(this, DOC_SCHEMA_VALIDATION)}>
-                Learn more about validations
-              </a>
-            </div>
-          </ZeroState>
-        </div>
-      );
+    if (!this.props.isZeroState) {
+      return;
     }
+
+    if (!this.props.isLoaded) {
+      // Don't display the form until we finished fetching validation because that would be misleading.
+      return;
```

Comment:
Maybe we can render the same loading state as Compass schema? (without the cancel button) 
https://github.com/mongodb-js/compass/blob/main/packages/compass-schema/src/components/steps/steps.jsx#L46
Like you said, it's probably rare a user would ever see this, but might be worth adding it until we have a generalized loading component. A user might see it on really slow internet? Not a blocker.